motion.c

```diff
@@ -1141,6 +1142,29 @@ static int motion_init(struct context *cnt)
         }
     }
 
+    /* Initialize 50% scaled substream server if substream port is specified to not 0
+       But only if dimensions are 8-modulo after scaling. Otherwise disable substream */
+    if (cnt->conf.substream_port){
+        if ((cnt->conf.width / 2) % 8 == 0  && (cnt->conf.height / 2) % 8 == 0
+        && cnt->imgs.type == VIDEO_PALETTE_YUV420P){
+            if (stream_init (&(cnt->substream), cnt->conf.substream_port, cnt->conf.stream_localhost,
+                cnt->conf.ipv6_enabled) == -1) {
+                MOTION_LOG(ERR, TYPE_ALL, SHOW_ERRNO, "Problem enabling motion-substream server in port %d",
+                           cnt->conf.substream_port);
+                cnt->finish = 1;
```

Comment:
I think we will need a `cnt->conf.substream_port = 0;` in here as identified in the recently closed issue.